sklearnex/ensemble/forest.py

```diff
@@ -16,3 +16,1137 @@
 #===============================================================================
 
 from daal4py.sklearn.ensemble import RandomForestClassifier, RandomForestRegressor
+
+from daal4py.sklearn._utils import (
+    daal_check_version, sklearn_check_version,
+    make2d, get_dtype
+)
+
+import numpy as np
+
+import numbers
+
+import warnings
+
+from abc import ABC
+
+from sklearn.exceptions import DataConversionWarning
+
+from .._config import get_config, config_context
+from .._device_offload import dispatch, wrap_output_data
+
+from sklearn.ensemble import ExtraTreesClassifier as sklearn_ExtraTreesClassifier
+from sklearn.ensemble import ExtraTreesRegressor as sklearn_ExtraTreesRegressor
+
+from sklearn.utils.validation import (
+    check_is_fitted,
+    check_consistent_length,
+    check_array,
+    check_X_y)
+
+from onedal.datatypes import _check_array, _num_features, _num_samples
+
+from sklearn.utils import check_random_state, deprecated
+
+from sklearn.base import clone
+
+from sklearn.tree import ExtraTreeClassifier, ExtraTreeRegressor
+from sklearn.tree._tree import Tree
+
+from onedal.ensemble import ExtraTreesClassifier as onedal_ExtraTreesClassifier
+from onedal.ensemble import ExtraTreesRegressor as onedal_ExtraTreesRegressor
+from onedal.primitives import get_tree_state_cls, get_tree_state_reg
+
+from scipy import sparse as sp
+
+if sklearn_check_version('1.2'):
+    from sklearn.utils._param_validation import Interval, StrOptions
+
+
+class BaseETForest(ABC):
```

Comment:
Does it mean BaseExtraTreesForest? I think it's better to name like BaseEnsemble or BaseTrees